napari/utils/settings/_manager.py

```diff
@@ -0,0 +1,131 @@
+"""Settings management.
+"""
+
+import os
+from pathlib import Path
+
+from pydantic import ValidationError
+from yaml import safe_dump, safe_load
+
+from napari.utils.settings._defaults import (
+    ApplicationSettings,
+    ConsoleSettings,
+    PluginSettings,
+)
+
+
+class SettingsManager:
+    """"""
+
+    def __init__(self, config_path=None):
+        self._config_path = Path.home() / ".napari" / "settings"
```

Comment:
yeah, i like `~/.config/napari`  unless we could think of other things to go in `~/.napari`

Reply:
There are packages to determine where to put config, that also support Env variable o decide. For example https://pypi.org/project/appdirs/.

Reply:
yeah, we do already depend on appdirs, so we can certainly use it ... but I believe there was some hesitation to put settings in those sorts of OS-specific places (in favor of a typical dotfiles).  I'm fine either way (appdirs, or .config), maybe @jni wants to weigh in here

Reply:
I kinda favor  `~/.napari`, not sure but I guess there will other things that make sense to put in there eventually 🤷🏼

Reply:
I very much like appdirs as the standard, which argues for ~/Library on macOS and ~/.config/napari on Linux. @tlambert03 perhaps you are remembering my hesitation to use QSettings, which indeed *also* takes care of these platform-specific issues. But I very much would like us to deal with those issues, and appdirs seems like a great, Qt-agnostic way to do it. I really dislike .napari, I think that pattern is used less and less since people have recognised that it leads to endless .whatevers in their home dirs.

Reply:
cool.  Then lets use appdirs!

Reply:
Fixing!